BaseClasses.py

```diff
@@ -592,7 +592,7 @@ def not_flooding_a_key(self, world, location):
             flood_location = world.get_location(flooded_keys[location.name], location.player)
             item = flood_location.item
             item_is_important = False if not item else item.advancement or item.bigkey or item.smallkey
-            return flood_location in self.locations_checked or not item_is_important
+            return flood_location in self.locations_checked or not item_is_important or location.parent_region.name != 'Swamp Trench 2 Alcove'
```

Comment:
There are 2 floodable keys, so you would need to check for both names.

Reply:
You're right!
Added the other location.